core/src/main/java/hudson/search/SuggestedItem.java

```diff
@@ -81,7 +81,7 @@ public String getUrl() {
     
     private static SuggestedItem build(SearchableModelObject searchContext, Item top) {
         ItemGroup<? extends Item> parent = top.getParent();
-        if (parent instanceof Item && parent != searchContext) {
+        if (parent instanceof Item) {
```

Comment:
Would be wise to add a test case for this too, since I can hardly guess what effect this patch will have on actual behavior without creating a build and trying all the combinations myself.